domain/src/test/java/org/mobilitydata/gtfsvalidator/domain/entity/gtfs/AgencyTest.java

```diff
@@ -131,4 +133,32 @@ public void createAgencyWithValidValuesShouldNotGenerateNotice() {
 
         assertTrue(entityBuildResult.getData() instanceof Agency);
     }
-}
\ No newline at end of file
+
+    @Test
+    void createAgencyWithBlankAgencyIdShouldGenerateInvalidAgencyIdNotice() {
+        final Agency.AgencyBuilder underTest = new Agency.AgencyBuilder();
+
+        final EntityBuildResult<?> entityBuildResult = underTest.agencyId(BLANK_STRING_VALUE)
+                .agencyName(STRING_TEST_VALUE)
+                .agencyUrl(STRING_TEST_VALUE)
+                .agencyTimezone(STRING_TEST_VALUE)
+                .agencyLang(STRING_TEST_VALUE)
+                .agencyPhone(STRING_TEST_VALUE)
+                .agencyFareUrl(STRING_TEST_VALUE)
+                .agencyEmail(STRING_TEST_VALUE)
+                .build();
+
+        assertTrue(entityBuildResult.getData() instanceof List);
+        // suppressed warning since cast check is not useful here: the test is designed so that method .getData()
+        // returns a list of notices.
+        //noinspection unchecked
+        final List<InvalidAgencyIdNotice> noticeCollection =
+                (List<InvalidAgencyIdNotice>) entityBuildResult.getData();
+        final InvalidAgencyIdNotice notice = noticeCollection.get(0);
+
```

Comment:
Same, I'm not sure this should be done in the entity as a blank or null ID should be valid for a sole agency.

Reply:
See https://github.com/MobilityData/gtfs-validator/pull/269#discussion_r437533627

Reply:
See https://github.com/MobilityData/gtfs-validator/pull/269#discussion_r443566311